### PR TITLE
Implement indeterminate icon for checkboxes

### LINKED ISSUE #27: Indeterminate state for checkboxes
Hey @adamwathan, thanks for the amazing library and this plugin!

I'm writing here so I mostly don't forget, but it seems that `.form-checkbox` removes native stylings but doesn't add one for the indeterminate state.

Checkboxes can be checked, unchecked, as well as indeterminate. Indeterminate is not usually used, but has its visual benefits. More info [on MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#Indeterminate_state_checkboxes).

Would be great if the `form-checkbox` class supports the indeterminate state.

### PR DESCRIPTION
**Pull request by swanson**
Adds an indeterminate state icon to checkboxes

image

The icon is this SVG that Adam / Steve provided: https://gist.github.com/adamwathan/77588f8c72d8f21d1022253a38a4fb03

Icon can be customized using an `indeterminateIcon` property and respects the `iconColor` customization as well.

Closes #27 Closes #62 Closes #50 (thanks for these PRs as they were a good starting point!)

### CODE REVIEW

File: src/index.test.js [hunk 1]
```diff
@@ -49,7 +49,9 @@ it('should generate the default classes for the form components', async () => {
 
     .form-input:focus {
       outline: none;
-      box-shadow: 0 0 0 var(--ring-offset-width, 0) var(--ring-offset-color, #fff), 0 0 0 calc(1px + var(--ring-offset-width, 0px)) var(--ring-color, #2563eb), var(--box-shadow, 0 0 #0000);
+      --ring-offset-shadow: 0 0 0 var(--ring-offset-width, 0) var(--ring-offset-color, #fff);
```
Comment (swanson) [inline src/index.test.js:52]: These snapshots appear to be broken on master (causing a failing build). Can rebase if it's been resolved.

Comment (adamwathan) [reply src/index.test.js:52]: Updated snapshots!

**Comment by adamwathan** [comment]
@swanson What's the reason for using `[indeterminate]` instead of the `:indeterminate` pseudo selector? I keep seeing both approaches in different articles online but can't find a good explanation for why you would use one vs. the other.

**Comment by swanson** [comment]
> @swanson What's the reason for using `[indeterminate]` instead of the `:indeterminate` pseudo selector? I keep seeing both approaches in different articles online but can't find a good explanation for why you would use one vs. the other.

I'm not sure, but he pseudo selector does match the conventions of the project better and seems equally "official" to me: https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate. The honest answer is that I was building off the other PRs and that's what they used :)

**Comment by adamwathan** [comment]
Haha gotcha! I have been working on this for the past little bit to see if there was an approach that didn't require a new key like indeterminateIcon and landed on this PR:

https://github.com/tailwindlabs/tailwindcss-custom-forms/pull/86

Going to go with that approach instead I think but really appreciate your work on this and also appreciate the nudge to just get this stupid feature done ❤️

**Pull request closed without merging**